src/osvr/USBSerial/USBSerialDevInfo_MacOSX.h

```diff
@@ -87,15 +85,62 @@ namespace usbserial {
 
             return (vendor_matches && product_matches);
         }
-
-    } // end namespace
+    }
 
     std::vector<USBSerialDevice>
     getSerialDeviceList(boost::optional<uint16_t> vendorID,
                         boost::optional<uint16_t> productID) {
 
-        //FIXME: STUB
         std::vector<USBSerialDevice> devices;
+
+        io_iterator_t serialPortIterator;
+        io_object_t serialPortService;
+        // find all serial ports
+        if (findSerialPorts(&serialPortIterator) == true) {
+            // iterate over serial ports, getting vid and pid
+            while ((serialPortService = IOIteratorNext(serialPortIterator)) !=
```

Comment:
If `IOIternatorNext(nullptr)` is well-defined, then this could be written as `io_iterator_t serialPortIterator = findSerialPorts(); while ((serialPortService = IOIteratorNext(serialPortIterator)) { ... }` if you adjust the `findSerialPorts()` function to return the iterator (and change any `return false;` statements to `return nullptr;`).